src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/LanguageServer/Handler/Restore/RestorePartialResult.cs

```diff
@@ -0,0 +1,13 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Serialization;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler;
+
+[DataContract]
+internal record RestorePartialResult(
+    [property: DataMember(Name = "stage")] string Stage,
```

Comment:
what does 'stage' mean here?

Reply:
Stage indicates what overall step is happening.  This value is shown in the progress bar (whereas the messages are only shown in the output window)

Reply:
i see.  can you doc.  It wasn't clear.  Looking at the PR it makes more sense now.  It allows the messages to be bucketed together right?  So you can tell which project these are associated with?